lib/install/phases/postsystem.go

```diff
@@ -150,29 +151,28 @@ type nodesExecutor struct {
 	Client *kubernetes.Clientset
 	// Application is the application being installed
 	Application app.Application
+	// Node is the node that should be labeled / tainted
+	Node storage.Server
 	// ExecutorParams is common executor params
 	fsm.ExecutorParams
 }
 
 // Execute executes the nodes phase
 func (p *nodesExecutor) Execute(ctx context.Context) error {
-	for _, server := range p.Plan.Servers {
-		p.Progress.NextStep("Updating node %v with labels and taints",
-			server.Hostname)
-		// find this node's profile
-		profile, err := p.Application.Manifest.NodeProfiles.ByName(server.Role)
-		if err != nil {
-			return trace.Wrap(err, "could not find node profile for %#v", server)
-		}
-		// update the node with labels and taints, try a few times to
-		// account for possible transient errors
-		err = utils.Retry(defaults.RetryInterval, defaults.RetryLessAttempts,
-			func() error {
-				return p.updateNode(p.Client, server, profile.Labels, profile.Taints)
-			})
-		if err != nil {
-			return trace.Wrap(err)
-		}
+	p.Progress.NextStep("Updating node %v with labels and taints", p.Node.Hostname)
+	// find this node's profile
+	profile, err := p.Application.Manifest.NodeProfiles.ByName(p.Node.Role)
+	if err != nil {
+		return trace.Wrap(err, "could not find node profile for %#v", p.Node)
+	}
+	// update the node with labels and taints, try a few times to
+	// account for possible transient errors
+	err = utils.Retry(defaults.RetryInterval, defaults.RetryLessAttempts,
```

Comment:
I think it's better to use exponential backoff when communicating with the API server